src/main/java/uk/gov/companieshouse/api/accounts/controller/SecretaryController.java

```diff
@@ -0,0 +1,125 @@
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
+import uk.gov.companieshouse.api.accounts.model.rest.DirectorsReport;
+import uk.gov.companieshouse.api.accounts.model.rest.directorsreport.Secretary;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.impl.SecretaryServiceImpl;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/directors-report/secretaries", produces = MediaType.APPLICATION_JSON_VALUE)
+public class SecretaryController {
+
+    @Autowired
+    private SecretaryServiceImpl secretaryService;
+
+    @Autowired
+    private ErrorMapper errorMapper;
+
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
+    @PostMapping
+    public ResponseEntity create(@Valid @RequestBody Secretary secretary, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId, HttpServletRequest request) {
+
+        if(bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+
+            ResponseObject<Secretary> response = secretaryService.create(secretary, transaction, companyAccountId, request);
+            return apiResponseMapper.map(response.getStatus(), response.getData(), response.getErrors());
+
+        } catch(DataException ex) {
+
+            LoggingHelper.logException(companyAccountId, transaction, "Failed to create secretary resource", ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+    }
+
+    @PutMapping("/{secretaryId}")
+    public ResponseEntity update(@Valid @RequestBody Secretary secretary, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId, @PathVariable("secretaryId") String secretaryId,
+                                 HttpServletRequest request) {
+
+        DirectorsReport directorsReport = (DirectorsReport) request.getAttribute(AttributeName.DIRECTORS_REPORT.getValue());
+        if(directorsReport.getLinks().get(SmallFullLinkType.DIRECTORS_REPORT.getLink()) == null &&
```

Comment:
`directorsReport.getLinks().get(SmallFullLinkType.DIRECTORS_REPORT.getLink()) == null` this check shouldn't be needed, nor will it ever return true